src/main/steps/existing-application/post.ts

```diff
@@ -25,24 +27,35 @@ export class ExistingApplicationPostController extends PostController<AnyObject>
       req.session.errors = form.getErrors(formData);
       if (req.session.errors.length === 0) {
         try {
+          const caseworkerUserApi = getCaseApi(await getSystemUser(), req.locals.logger);
+          const existingCase = await caseworkerUserApi.getCaseById(req.session.existingCaseId);
           if (formData.existingOrNewApplication === existingOrNew.Existing) {
-            req.locals.logger.error(
-              `UserId: "${req.session.user.id}" has chosen to continue with existing application: ${req.session.existingCaseId}
-              and cancelling case invite: ${req.session.inviteCaseId}`
+            req.locals.logger.info(
+              `UserId: ${req.session.user.id} has chosen to continue with existing application: ${req.session.existingCaseId}
+                    and cancelling case invite: ${req.session.inviteCaseId}`
             );
-            const caseworkerUserApi = getCaseApi(await getSystemUser(), req.locals.logger);
             await caseworkerUserApi.triggerEvent(req.session.inviteCaseId, {}, SYSTEM_CANCEL_CASE_INVITE);
 
-            req.session.userCase = await caseworkerUserApi.getCaseById(req.session.existingCaseId);
+            req.session.userCase = existingCase;
             req.session.isApplicant2 = await caseworkerUserApi.isApplicant2(
               req.session.userCase.id,
               req.session.user.id
             );
-
             nextUrl = HOME_URL;
           } else {
-            req.session.applicantChoosesNewInviteCase = true;
-            nextUrl = `${APPLICANT_2}${ENTER_YOUR_ACCESS_CODE}`;
+            if (this.isAllowedToLinkToNewCase(existingCase)) {
```

Comment:
is this meant to take in existing case?

Reply:
yes, because we check if their existing case is submitted/aosSubmitted, if it is, we don't let them leave it basically

Reply:
i think the name might need changing. to me it reads as if the function is checking the case to see if the user is allowed to linked to it. maybe change to isAllowedToUnlinkExisingCase or isAllowedToLeaveExisingCase??

Reply:
done